mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/struct/FVecFactory.scala

```diff
@@ -0,0 +1,51 @@
+package ml.combust.mleap.xgboost.runtime.struct
+
+import java.{lang, util}
+
+import biz.k11i.xgboost.util.FVec
+import ml.combust.mleap.tensor.{SparseTensor, Tensor}
+import org.apache.spark.ml.linalg.{DenseVector, SparseVector}
+
+import scala.collection.JavaConversions.mapAsJavaMap
+
+
+object FVecFactory {
+  private def toJavaMap(map: Map[Int, Float]): util.Map[lang.Integer, lang.Float] = {
+    mapAsJavaMap(map).asInstanceOf[util.Map[lang.Integer, lang.Float]]
+  }
+
+  /**
+    *  NOTE: all these methods cast doubles to floats, because doubles result in compounding differences
```

Comment:
@ancasarb reference to this comment regarding floats vs double bug

Reply:
👍